### PR TITLE
Added policy for AMA upgrade

### PR DESCRIPTION
**Pull request by QBY-MarkusMaring**
# Description

This PR adds a policy set which targets VMs, VMSSs and Arc machines. For all these resources the following happens:

- Installation of AMA agent
- Installation of Dependency Agent
- Enable System Assigned managed identity needed for monitoring
- Associate machines with the central DCR for VMInsights
- Associate machines with additional DCRs as specified from parameters

Where possible Microsoft built-in policies were used. These policies include a filter written by Microsoft that determines if a OS is supported. This filter can be disabled if needed, but can't be edited. 

Our custom policies use a more advanced filter that can be set using parameters. The default behavior when no parameters are set will not filter at all and will instead attempt to deploy the agent on every machine. 

In the future we could consider using only custom policies with this advanced filter.

This PR will also disable the previous policy which installed the old Agent (MMA).

# Change overview (tick true):

- [x] This introduces backward incompatible changes
- [ ] This adds a new backward compatible Feature
- [ ] This fixes a Bug

# Version information: 

- Previous Version: `2.3.0`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `3.0.0` (once we release it)

# How Has This Been Tested?

- [x] Deployment test via terraform and caf
- [x] Deployment test with additional DCRs
- [x] Policy test for osOffer filter
- [x] Policy test for osSKUs filter
- [x] Policy test for virtual machines
- [x] Policy test for virtual machine scale sets
- [ ] Policy test for Arc machines

# Checklist:

- [x] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [x] I have updated the documentation
- [x] I have updated the CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,14 @@ and this module adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+# [3.0.0] - 2024-03-01
```
Comment (QBY-ThimoLimpert) [inline CHANGELOG.md:10]:
```suggestion
```
I think we should wait for Update Manager and tags before creating a release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,10 +7,14 @@ and this module adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.
...
+
+- Added initiative for new AMA monitoring
```
Comment (QBY-ThimoLimpert) [inline CHANGELOG.md:12]:
```suggestion
### Added
- Initiative deploying new Azure Monitoring Agent and associating data collection rules

### Removed
- Policies deploying the old Microsoft Monitoring Agent
```

File: archetypes/archetype_definition_qby_msp.json [hunk 3]
```diff
@@ -1,10 +1,9 @@
 {
     "qby_msp": {
         "policy_assignments": [
-            "QBY-Deploy-VM-Backup",
             "QBY-Deploy-DomainJoin",
-            "Deploy-VM-Monitoring",
-            "QBY-Deploy-LAW-Agent"            
+            "QBY-Deploy-VM-Backup",
```
Comment (QBY-ThimoLimpert) [inline archetypes/archetype_definition_qby_msp.json:5]: We only monitor and backup VMs, that are managed by us? :O

File: policy_definitions/vmmonitoring/policy_assignment_qby_deploy_vm_monitoring.tmpl.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "QBY-Deploy-VM-Monitoring",
+  "type": "Microsoft.Authorization/policyAssignments",
+  "apiVersion": "2019-09-01",
+  "properties": {
+    "description": "Deploys azure monitor agent and dependency agent to linux and windows VMs, VMSS and Arc machines. Associates machines to all default data collection rules.",
+    "displayName": "QBY-Deploy-VM-Monitoring",
```
Comment (QBY-ThimoLimpert) [inline policy_definitions/vmmonitoring/policy_assignment_qby_deploy_vm_monitoring.tmpl.json:7]:
```suggestion
    "displayName": "Deploy VM agents and DCR associations",
```

**Review by QBY-ThimoLimpert — CHANGES_REQUESTED** [review]
Only minor improvements. Overall looking good!

File: archetypes/archetype_definition_qby_msp.json (hunk 3 above)
Comment (QBY-MarkusMaring) [reply archetypes/archetype_definition_qby_msp.json:5]: Yes. That is the default. If the customer want's this too we will gladly deploy the policies in their MGs also

**Review by QBY-ThimoLimpert — APPROVED** [review]

**Pull request merged**